src/Microsoft.AspNetCore.ResponseCaching/Internal/ResponseCachingKeyProvider.cs

```diff
@@ -124,9 +124,12 @@ public string CreateStorageVaryByKey(ResponseCachingContext context)
                             .Append(header)
                             .Append("=");
 
-                        for (var j = 0; j < headerValues.Count; j++)
+                        var headerValuesArray = headerValues.ToArray();
+                        Array.Sort(headerValuesArray, StringComparer.Ordinal);
```

Comment:
Anyway we can save the sorted headerValues array if they haven't been modified? Though don't think it will matter much because CreateStorageVaryByKey is called usually once per request.

Reply:
Also can we check if the size is 1 before we sort? Idk if it will have meaningful perf impact.

Reply:
Right it's called per request and per cached response so saving it won't be useful. I have tried checking the size and special casing 0 or 1 element. I didn't notice any performance difference and reverted the change to keep things clean. Seems like ToArray and Array.Sort handle these cases pretty well.